src/Components/Server/src/DependencyInjection/ComponentServiceCollectionExtensions.cs

```diff
@@ -19,19 +21,42 @@ namespace Microsoft.Extensions.DependencyInjection
     public static class ComponentServiceCollectionExtensions
     {
         /// <summary>
-        /// Adds Razor Component app services to the service collection.
+        /// Adds Server-Side Blazor services to the service collection.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/>.</param>
-        /// <returns>The <see cref="IServiceCollection"/>.</returns>
-        public static IServiceCollection AddServerSideBlazor(this IServiceCollection services)
+        /// <returns>An <see cref="IServerSideBlazorBuilder"/> that can be used to further customize the configuration.</returns>
+        public static IServerSideBlazorBuilder AddServerSideBlazor(this IServiceCollection services)
         {
-            services.AddSignalR()
-                .AddHubOptions<ComponentHub>(options =>
+            return AddServerSideBlazor(services, configure: null);
+        }
+
+        /// <summary>
+        /// Adds Server-Side Blazor services to the service collection.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+        /// <param name="configure">A delegate used to the configure the <see cref="HubOptions"/> that affect Server-Side Blazor.</param>
+        /// <returns>An <see cref="IServerSideBlazorBuilder"/> that can be used to further customize the configuration.</returns>
+        public static IServerSideBlazorBuilder AddServerSideBlazor(this IServiceCollection services, Action<HubOptions> configure)
```

Comment:
I provided `AddServerSideBlazor(... Action<HubOptions>)` because that was in the notes I had sent out.

Hearing your concern, I agree with it. I'll take it out, and users will have to `AddServerSideBlazor().AddHubOptions()`. This way if we add an options type specific to us, we have a place to put it.